feugy/tabulous
# chore(web): updates to latest Sveltekit conventions

### :book: What's in there?

Some time ago, the SvelteKit team announced [a migration guide](https://github.com/sveltejs/kit/discussions/5774) for the latest versions.
This PR applies them, allowing us to stay on the edge.

### :test_tube: How to test?

The integration tests covers the login/logout and authentication enforcement code.
Check out the [preview deployment](https://tabulous-git-chore-updates-sveltekit-feugy.vercel.app/home)
1. browse to /
   > you should be redirected to home, anonymous access
1. starts a new game
   > you should be redirected to login page, with the correct `redirect`  query parameter
1. enter invalid credentials
   > an error message is displayed
1. enter valid credentials
   > access is granted, and the new game is started.
1. refresh the page
   > game is displayed, _copy the page URL_
1. open game menu and leave page
   > home page is displayed, with started games and authenticated catalog
1. click logout button
   > home page is refreshed, displaying anonymous catalog
1. browse to the game URL
   > login page is displayed, with correct `redirect` query parameter.

### :up: Notes to reviewers

Files have been mainly renamed:
- `routes/index.svelte` > `routes/+page.svelte`
- `routes/__layout.svelte` > `routes/+layout.svelte`
- `routes/__layout-auth@default.svelte` > `routes/+layout-auth@default.svelte` + `routes/+layout-auth.server.js` 
   > the svelte layout inherits from default layout, but it's not the case of its server counterpart. Hence why it's explicitly calling the load function.

More counter-intuitive changes:
- `hooks.js`'s `getSession()`  is now `routes/+layout.server.js`: since session [makes no sense](https://github.com/sveltejs/kit/discussions/5883) in this new design. Hence why some pages are now using data loaded from server.
- `routes/logout.js` is implemented inside `hooks.js`, because `routes/logout/+server.s` could not reset the cookie: its returned response would not be mutable from `hooks.js`'s `setCookie()`
